### PR TITLE
add a short explaining about plugins usages

### PR DESCRIPTION
**Pull request by jiraguha**
Solving #7030

### CODE REVIEW

File: test_plugin/README.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+name = "test_plugin"
+...
+
+[lib]
+crate-type = ["cdylib"]
+
+[dependencies]
+...
+
+deno_core = { path = "../core" }
```
Comment (nayeemrmn) [inline test_plugin/README.md:40]: This is not how you'd usually add `deno_core`.

Comment (jiraguha) [reply test_plugin/README.md]: yep, outside the deno project we rely on the Rust community’s crate registry! Change done

File: test_plugin/README.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+$ cargo build -p test_plugin --release
+```
+
+## In js
+
+#### 5- open your lib with `Deno.openPlugin`. You had to specify the build target from (4)
+
+```js
+// in test.js
+ const rid = Deno.openPlugin(<path to *.dylib>);
```
Comment (divy-work) [inline test_plugin/README.md:82]: The lib does not end with a `.dylib` always.

Windows: `.dll`
Linux: `.so`
MacOS: `.dylib`

File: test_plugin/README.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+test_plugin
+├── Cargo.toml
+├── src
+│   └── lib.rs
+└── tests
+    ├── ...
+    └── test.js
+```
+
+It is a rust repo.
```
Comment (divy-work) [inline test_plugin/README.md:21]: Should this be specified?

File: test_plugin/README.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+deno_core = "0.52.0"
+...
+```
+
+#### 2- wrap your custom lib functionalities in ops functions,
+
+```rust
+// in lib.rs
+fn op_test_sync(
```
Comment (divy-work) [inline test_plugin/README.md:48]: Should also cover `op_test_async`

File: test_plugin/README.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+```
+
+#### 3- register then inside the plugin container `deno_plugin_init`.
+
+```rust
+// in lib.rs
+#[no_mangle]
+pub fn deno_plugin_init(interface: &mut dyn Interface) {
+  interface.register_op("testSync", op_test_sync);
+  ...
```
Comment (divy-work) [inline test_plugin/README.md:66]: `testAsync`

File: test_plugin/README.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+}
+```
+
+#### 4- build your lib with cargo : `cargo build -p <name of the crate>`
+
+```shell
+$ cargo build -p test_plugin --release
```
Comment (divy-work) [inline test_plugin/README.md:73]: just `cargo build --release` should be enough

**Comment by ebebbington** [comment]
I think it's a really good idea to actually explain how to create a simple test plugin, what it uses, how it works, what these methods do etc. Then a user can use this information to be confident enough to build whatever they want

**Comment by bartlomieju** [comment]
@jiraguha thank you for the PR, but I'm gonna close it without a merge. In light of #8490 it's not yet clear if we want to add documentation for plugins or remove them.

**Pull request closed without merging**